packages/augur-core/src/contracts/libraries/math/SafeMathInt256.sol

```diff
@@ -71,4 +71,17 @@ library SafeMathInt256 {
     function fxpDiv(int256 a, int256 b, int256 base) internal pure returns (int256) {
         return div(mul(a, base), b);
     }
+
+    function sqrt(int256 y) internal pure returns (int256 z) {
+        if (y > 3) {
+            int256 x = (y + 1) / 2;
+            z = y;
+            while (x < z) {
+                z = x;
+                x = (y / x + x) / 2;
+            }
+        } else if (y != 0) {
+            z = 1;
+        }
+    }
```

Comment:
Recommend a test suite for this function specifically.  My confidence in it being totally correct is very low.  Should be easy to generate a whole bunch of test cases, with a focus around boundary conditions (e.g., right before, after, and on a rounding condition like `sqrt(15)`, `sqrt(16)`, `sqrt(17)`.  Also would be good to text whatever the largest int256 square is.

Note: I don't think we need negative numbers here, and that adds complexity so I would recommend just having this be `uint256` instead.

Reply:
Wow yeah this and the SafeMathUint256 sqrt are untested. Will add!